### PR TITLE
[AML] compile fix

### PR DESCRIPTION
**Pull request by peak3d**
## Description
Fix compile issues after RenderManager rework

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp [hunk 1]
```diff
@@ -2257,8 +2257,8 @@ std::string CAMLCodec::GetStereoMode()
     default:                                  stereo_mode = m_hints.stereo_mode; break;
   }
 
-  if(m_processInfo.GetVideoSettings().m_StereoInvert)
-    stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
+  if (m_processInfo.GetVideoSettings().m_StereoInvert)
+    ;//stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
```
Comment (peak3d) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: @FernetMenta I have no idea how to implement this one

Comment (codesnake) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: @peak3d My implementation:
```
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -2248,18 +2248,26 @@ bool CAMLCodec::SetVideo3dMode(const int mode3d)

 std::string CAMLCodec::GetStereoMode()
 {
-  std::string  stereo_mode;
+  std::string stereoMode;

   switch(m_processInfo.GetVideoSettings().m_StereoMode)
   {
-    case RENDER_STEREO_MODE_SPLIT_VERTICAL:   stereo_mode = "left_right"; break;
-    case RENDER_STEREO_MODE_SPLIT_HORIZONTAL: stereo_mode = "top_bottom"; break;
-    default:                                  stereo_mode = m_hints.stereo_mode; break;
+    case RENDER_STEREO_MODE_SPLIT_VERTICAL:
+      stereoMode = "left_right";
+      if (m_processInfo.GetVideoSettings().m_StereoInvert)
+        stereoMode = "right_left";
+      break;
+    case RENDER_STEREO_MODE_SPLIT_HORIZONTAL:
+      stereoMode = "top_bottom";
+      if (m_processInfo.GetVideoSettings().m_StereoInvert)
+        stereoMode = "bottom_top";
+      break;
+    default:
+      stereoMode = m_hints.stereo_mode;
+      break;
   }

-  if(m_processInfo.GetVideoSettings().m_StereoInvert)
-    stereo_mode = RenderManager::GetStereoModeInvert(stereo_mode);
-  return stereo_mode;
+  return stereoMode;
 }
```

Comment (peak3d) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: thanx @codesnake !

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: @codesnake your code is not correct. A stereo mode set by decoder is not supposed to get overridden. see VPV

Comment (peak3d) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: @FernetMenta does this mean if m_hints.stereo_mode is valid is has to be uset with priority?

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: no, check vpv
- if decoder sets stereo mode, use this
- if not set by decoder use udes setting
- if none of the above applies, se hints

**Comment by peak3d** [comment]
@codesnake does the AML decoder provide any information about the stereo-mode?

**Comment by codesnake** [comment]
@peak3d AFAIK no.

**Comment by peak3d** [comment]
int codec_stereo(codec_para_t *p); in aml_libs, could this be the one?

**Comment by codesnake** [comment]
@peak3d codec_stereo() is for audio.

**Comment by peak3d** [comment]
ok :-) and this one?
amcodec/include/amports/amstream.h:#define  AMSTREAM_IOC_GET_3D_TYPE  _IOW((AMSTREAM_IOC_MAGIC), 0x3d, unsigned int)

**Comment by codesnake** [comment]
@peak3d It seems AMSTREAM_IOC_GET_3D_TYPE returns 3D mode currently set. But there's also AMSTREAM_IOC_GET_SOURCE_VIDEO_3D_TYPE. It looks like what we need.

**Comment by codesnake** [comment]
@peak3d Don't forget to remove memset() for m_videobuffer in CDVDVideoCodecAmlogic::Open(). 
VideoPicture::stereoMode is now std::string and cannot be simply zeroed. Otherwise the decoder will crash in GetPicture(). See https://github.com/codesnake/xbmc/commit/c07b8c8ad2eabb187d84a261831d2d80dc329810#diff-a2a2ab186a31ae40a3b2985a6828e8c9L284.

**Comment by peak3d** [comment]
ok

**Comment by MrMC** [comment]
Really should add a reset() to VideoPicture. And why does VideoPicture not follow naming convention ? Should be CVideoPicture as it is a class.

**Comment by peak3d** [comment]
@MrMC yes, will add an reset(), naming conventions of this class (which have been  a struct earlier iirc) is @FernetMenta ' working place.

**Comment by peak3d** [comment]
Removed memset of VideoPicture (which is a class now) in AndroidMediaCodec Decoder.

**Comment by peak3d** [comment]
@FernetMenta any objections?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp [hunk 2]
```diff
@@ -42,6 +42,37 @@ VideoPicture::~VideoPicture()
   }
 }
 
+void VideoPicture::Reset()
+{
+  videoBuffer = nullptr;
+  pts = AV_NOPTS_VALUE;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp]: DVD_NOPTS_VALUE

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp [hunk 3]
```diff
@@ -42,6 +42,37 @@ VideoPicture::~VideoPicture()
...
+  dts = AV_NOPTS_VALUE;
+  iFlags = 0;
+  iRepeatPicture = 0;
+  iDuration = 0;
+  iFrameType = 0;
+  color_space = AVCOL_SPC_UNSPECIFIED;
+  color_range = 1;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.cpp]: default is 0

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h [hunk 4]
```diff
@@ -53,6 +53,7 @@ struct VideoPicture
   ~VideoPicture();
   VideoPicture& CopyRef(const VideoPicture &pic);
   VideoPicture& SetParams(const VideoPicture &pic);
+  void Reset(); // reinitialize members, videoBuffer will NOT released if set!
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h]: a clean reset should also release videoBuffer

**Comment by FernetMenta** [comment]
You forgot to squash fixup commits

**Comment by Rechi** [comment]
squashed with github, but preferred would be squash locally and then force push as it would reflect that in  the PR too

**Comment by peak3d** [comment]
No, the first one was squashed, but unfortunately with an bug :-(

**Comment by FernetMenta** [comment]
You squashed vp changes into an aml fix :(

**Comment by FernetMenta** [comment]
Please revert and do again

**Comment by peak3d** [comment]
https://github.com/xbmc/xbmc/pull/13461

**Pull request merged**